c-deps: support 16K pages for ARM64

RaduBerinde opened the pull request:
We have had two reports of users not being able to run cockroach 23.1 on linux/arm64 with 16k pages (jemalloc does not support the page size). This change attempts to fix this by compiling jemalloc with 16K page support, only in the linux/arm64 configuration.

Epic: None
Informs: #106745
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106929)
<!-- Reviewable:end -->

rickystewart reviewed (approved):
This is fine but we need to re-build the pre-built `c-deps`. I'll wait until #106642 is merged to do that because we'll get conflicts otherwise. This can merge in the meantime (or you can wait until #106642 is merged, if you want to get everything in in one big commit).

Either way can you make sure to backport to 23.1?

RaduBerinde commented:
Thanks @rickystewart! Yes, I plan to backport to 23.1.

bors r+

RaduBerinde committed:
c-deps: support 16K pages for ARM64

We have had two reports of users not being able to run cockroach 23.1
on linux/arm64 with 16k pages (jemalloc does not support the page
size).  This change attempts to fix this by compiling jemalloc with
16K page support, only in the linux/arm64 configuration.

Epic: None
Informs: #106745
Release note: None

RaduBerinde commented:
bors r+

RaduBerinde commented:
blathers backport 23.1
